pkg/configs/configs.go

```diff
@@ -26,36 +27,57 @@ const (
 	RuleFormatV2 RuleFormatVersion = iota
```

Comment:
I understand that the iota/default value when marshalling/unmarshalling is nice, but it is a bit confusing when implementing an external service that we need to specify `rule_format_version: 1` when using prometheus version 2 rules. I am curious what others think.

Reply:
That sounds good to me, would it be possible to just add it to the custom marshal/unmarshal implementations for `Config`?

Reply:
It'd have to be defined on `RuleFormatVersion` to be safe, since we do things like unmarshaling into subfields of `Config` directly: https://github.com/weaveworks/cortex/pull/789/files#diff-ff5adb3414291874684b88cf594450acR247